fireplace/cards/utils.py

```diff
@@ -39,52 +41,40 @@
 
 
 # take a list of weights and a list of card pools and produce a random weighted sample without replacement
-def weighted_card_choice(source, weights, card_sets, count):
+# len(weights) == len(card_sets) (one weight per card set)
+def weighted_card_choice(source, weights: List[int], card_sets: List[str], count: int):
 
 	# special case for single weight (generic random pick)
 	if len(card_sets)==1:
 		return [source.controller.card(card, source=source) for card in random.sample(card_sets[0], count)]
 
-	# get all the cards matching each set of filters and create the weight line
-	# (user does not have to normalize weighting from 0-1)
-	cardlist = []
-	weightedsegmentstarts = [0] * len(card_sets)
-	weightlinelength = 0
+	# special case for all weights equal
+	if weights.count(weights[0]) == len(weights):
```

Comment:
This might be more idiomatic Python:

`if all(x==weights[0] for x in weights):`